sklearn/covariance/outlier_detection.py

```diff
@@ -36,7 +37,12 @@ class OutlierDetectionMixin(object):
     """
     def __init__(self, contamination=0.1):
         self.contamination = contamination
+        self.threshold_ = None
+        # FIXME to be removed in 0.18
         self.threshold = None
+        warnings.warn(("The threshold attribute is renamed to threshold_ from "
```

Comment:
Warnings should only be given to users who care. Your current approach warns all users, and breaks the code of the people that use ``threshold``.
You should add a ``@property def threshold(self): return self.threshold_`` that raises a warning, so if a user uses the old name, they are warned.